src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/service/MappingService.kt

```diff
@@ -63,6 +69,28 @@ class MappingService(
       ?: throw NotFoundException("prison id=$prisonId, nomis room id=$nomisRoomDescription")
 
   suspend fun deleteVisitMappings() = visitIdRepository.deleteAll()
+
+  fun getVisitMappingsByMigrationId(pageRequest: Pageable, migrationId: String): Mono<Page<MappingDto>> {
+    return visitIdReactiveRepository.findAllByLabelAndMappingTypeOrderByLabelDesc(
+      label = migrationId,
+      MappingType.MIGRATED,
+      pageRequest
+    ).collectList()
+      .zipWith(visitIdReactiveRepository.countAllByLabelAndMappingType(migrationId, mappingType = MappingType.MIGRATED))
+      .map { t ->
```

Comment:
separate count query required - int test checks the counts match the content in the response